src/modules/hip/kernel/hue.cpp

```diff
@@ -123,200 +174,325 @@ extern "C" __global__ void huergb_pkd(     unsigned char *input,
-                                     int *yroi_end,
-                                     unsigned int *height,
-                                     unsigned int *width,
-                                     unsigned int *max_width,
-                                     unsigned long *batch_index,
-                                     unsigned int *inc, // use width * height for pln and 1 for pkd
-                                    const int plnpkdindex // use 1 pln 3 for pkd
-                                    )
+extern "C" __global__ void saturation_batch(unsigned char *input,
+                                             unsigned char *output,
+                                             float *sat,
+                                             unsigned int *xroi_begin,
+                                             unsigned int *xroi_end,
+                                             unsigned int *yroi_begin,
+                                             unsigned int *yroi_end,
+                                             unsigned int *height,
+                                             unsigned int *width,
+                                             unsigned int *max_width,
+                                             unsigned long long *batch_index,
+                                             unsigned int *inc, // use width * height for pln and 1 for pkd
+                                            const int plnpkdindex) // use 1 pln 3 for pkd
 {
-  int id_x = hipBlockIdx_x *hipBlockDim_x + hipThreadIdx_x, id_y = hipBlockIdx_y *hipBlockDim_y + hipThreadIdx_y, id_z = hipBlockIdx_z *hipBlockDim_z + hipThreadIdx_z;
-  if (id_x >= width[id_z] || id_y >= height[id_z]) return;
-  uchar4 pixel; float4 hsv;
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if (id_x >= width[id_z] || id_y >= height[id_z])
+    {
+        return;
+    }
 
+    uchar4 pixel; float4 hsv;
+    int pixIdx = batch_index[id_z] + (id_y * max_width[id_z] + id_x) * plnpkdindex;
+    pixel.x = input[pixIdx];
+    pixel.y = input[pixIdx + inc[id_z]];
+    pixel.z = input[pixIdx + 2 * inc[id_z]];
+    pixel.w = 0.0;
 
-  int pixIdx = batch_index[id_z] + (id_y * max_width[id_z] + id_x) * plnpkdindex;
-  pixel.x = input[pixIdx];
-  pixel.y = input[pixIdx + inc[id_z]];
-  pixel.z = input[pixIdx + 2 * inc[id_z]];
-  pixel.w = 0.0;
+    if ((id_y >= yroi_begin[id_z]) && (id_y <= yroi_end[id_z]) && (id_x >= xroi_begin[id_z]) && (id_x <= xroi_end[id_z]))
+    {
+        hsv = convert_one_pixel_to_hsv(pixel);
+        hsv.y *= sat[id_z];
 
-  if ((id_y >= yroi_begin[id_z]) && (id_y <= yroi_end[id_z]) &&
-    (id_x >= xroi_begin[id_z]) && (id_x <= xroi_end[id_z])) {
-    hsv = convert_one_pixel_to_hsv(pixel);
-    hsv.y *= sat[id_z];
-    if (hsv.y > 1.0) {
-      hsv.y = 1.0;
+        if (hsv.y > 1.0)
```

Comment:
better to use min and max instruction here to avoid conditional branch

Reply:
Modified to use HIP's fmaxf, fminf